pkg/config/tlscfg/cert_watcher.go

```diff
@@ -60,27 +53,33 @@ func newCertWatcher(opts Options, logger *zap.Logger) (*certWatcher, error) {
 		cert = &c
 	}
 
-	watcher, err := fswatcher.NewWatcher()
-	if err != nil {
-		return nil, err
-	}
-
 	w := &certWatcher{
-		opts:    opts,
-		logger:  logger,
-		cert:    cert,
-		watcher: watcher,
+		opts:   opts,
+		logger: logger,
+		cert:   cert,
 	}
 
-	if err := w.setupWatchedPaths(); err != nil {
-		watcher.Close()
+	if err := w.watchCertPair(); err != nil {
+		return nil, err
+	}
+	if err := w.watchCert(w.opts.CAPath, rootCAs); err != nil {
 		return nil, err
 	}
+	if err := w.watchCert(w.opts.ClientCAPath, clientCAs); err != nil {
+		return nil, err
+	}
+
 	return w, nil
 }
 
 func (w *certWatcher) Close() error {
-	return w.watcher.Close()
+	var errs []error
+	for _, w := range w.watchers {
+		if err := w.Close(); err != nil {
+			errs = append(errs, err)
+		}
```

Comment:
you can actually fool the code coverage here by always appending error, because Join ignores nils
```suggestion
		errs = append(errs, w.Close())
```